HLTrigger/Muon/plugins/HLTMuonL3orL2PreFilter.cc

```diff
@@ -0,0 +1,349 @@
+      max_Dz(iConfig.getParameter<double>("MaxDz")),
+      min_DxySig(iConfig.getParameter<double>("MinDxySig")),
+      nsigma_Pt(iConfig.getParameter<double>("NSigmaPt")),
+      max_PtDifference(iConfig.getParameter<double>("MaxPtDifference")),
+      min_TrackPt(iConfig.getParameter<double>("MinTrackPt")),
+      max_NormalizedChi2(iConfig.getParameter<double>("MaxNormalizedChi2")),
+      min_Pt(iConfig.getParameter<double>("MinPt")),
+      max_Eta(iConfig.getParameter<double>("MaxEta")),
+      min_NmuonHits(iConfig.getParameter<int>("MinNmuonHits")),
+      min_Nhits(iConfig.getParameter<std::vector<int>>("MinNhits")),
+      absetaBins(iConfig.getParameter<std::vector<double>>("AbsEtaBins")),
+      minNstations(iConfig.getParameter<std::vector<int>>("MinNstations")),
+      minNchambers(iConfig.getParameter<std::vector<int>>("MinNchambers")) {}
+
+HLTMuonL3orL2PreFilter::HLTMuonL3orL2PreFilter(const edm::ParameterSet& iConfig)
+    : HLTFilter(iConfig),
+      L3candTag_(iConfig.getParameter<edm::InputTag>("L3CandTag")),
+      L3candToken_(consumes(L3candTag_)),
+      L2CandTag_(iConfig.getParameter<edm::InputTag>("L2CandTag")),
+      L2CandToken_(consumes(L2CandTag_)),
+      previousCandTag_(iConfig.getParameter<edm::InputTag>("PreviousCandTag")),
+      previousCandToken_(consumes(previousCandTag_)),
+      beamspotTag_(iConfig.getParameter<edm::InputTag>("BeamSpotTag")),
+      beamspotToken_(consumes(beamspotTag_)),
+      matchPreviousCand_(iConfig.getParameter<bool>("MatchToPreviousCand")),
+      maxDeltaR2_L2L3_(iConfig.getParameter<double>("MaxDeltaR2L2L3")),
+      cutParsL3_(iConfig.getParameter<edm::ParameterSet>("L3CandSelection")),
+      cutParsL2_(iConfig.getParameter<edm::ParameterSet>("L2CandSelection")) {
+  // check consistency of parameters
+  if (cutParsL2_.absetaBins.size() != cutParsL2_.minNstations.size() ||
+      cutParsL2_.absetaBins.size() != cutParsL2_.minNchambers.size() ||
+      cutParsL2_.absetaBins.size() != cutParsL2_.min_Nhits.size()) {
+    throw cms::Exception("Configuration")
+        << "error in ParameterSet \"L2CandSelection\": size of \"AbsEtaBins\" (" << cutParsL2_.absetaBins.size()
+        << "), \"MinNstations\" (" << cutParsL2_.minNstations.size() << "), \"MinNchambers\" ("
+        << cutParsL2_.minNchambers.size() << ") and \"MinNhits\" (" << cutParsL2_.min_Nhits.size() << ") differ";
+  }
+  if (cutParsL3_.absetaBins.size() != cutParsL3_.minNstations.size() ||
+      cutParsL3_.absetaBins.size() != cutParsL3_.minNchambers.size() ||
+      cutParsL3_.absetaBins.size() != cutParsL3_.min_Nhits.size()) {
+    throw cms::Exception("Configuration")
+        << "error in ParameterSet \"L3CandSelection\": size of \"AbsEtaBins\" (" << cutParsL3_.absetaBins.size()
+        << "), \"MinNstations\" (" << cutParsL3_.minNstations.size() << "), \"MinNchambers\" ("
+        << cutParsL3_.minNchambers.size() << ") and \"MinNhits\" (" << cutParsL3_.min_Nhits.size() << ") differ";
+  }
+}
```

Comment:
```suggestion

  if(maxDeltaR_L2L3_ <= 0.) {
    throw cms::Exception("Configuration") << "invalid value for parameter \"MaxDeltaRL2L3\" (must be > 0): " << maxDeltaR_L2L3_;
  }
}
```
It's more intuitive/convenient to use DeltaR as parameter in the configuration. The conversion to DeltaR2 can be done inside the plugin. Since a negative value does not make sense in this plugin (if i understand correctly), an exception can be thrown in that case in the ctor.